src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/executors/spark/SparkExecutor.scala

```diff
@@ -50,8 +50,9 @@ object SparkExecutor extends Executor[SparkInstanceParameters] {
       val submitTimeOut = SparkExecutorConfig.getSubmitTimeOut
       val latch = new CountDownLatch(1)
       val sparkAppHandle = getSparkLauncher(id, ji.jobName, jobParameters).startApplication(new SparkAppHandle.Listener {
+        import scala.math.Ordered.orderingToOrdered
         override def stateChanged(handle: SparkAppHandle): Unit =
-          if (handle.getState == SparkAppHandle.State.SUBMITTED) {
+          if (handle.getState >= SparkAppHandle.State.SUBMITTED) {
```

Comment:
How do you know the order of statuses? 
SparkAppHandle.State does not implement Ordered interface.

Reply:
I tested it and it works as expected